core/functional_tests/websocket/service.cpp

```diff
@@ -90,10 +90,37 @@ class WebsocketsFullDuplexHandler final
   }
 };
 
+class WebsocketsFullDuplexHandlerAlt final
+    : public server::websocket::WebsocketHandlerBase {
+ public:
+  static constexpr std::string_view kName = "websocket-duplex-handler-alt";
+
+  using WebsocketHandlerBase::WebsocketHandlerBase;
+
+  void Handle(server::websocket::WebSocketConnection& chat,
+              server::request::RequestContext&) const override {
+
+    server::websocket::Message message;
+    while (!engine::current_task::ShouldCancel()) {
+      const bool msgIsReceived = chat.Recv2(message);
+      if (msgIsReceived) {
+        if (message.close_status) break;
+        chat.Send(message);
```

Comment:
Use `SendBinary` here to make Python recive it as a binary data. Or change asserts in Pyhton to comaper text: `assert msg == b'B' ` to `assert msg == 'B' `